src/azure-cli-testsdk/azure/cli/testsdk/utilities.py

```diff
@@ -198,3 +199,18 @@ def process_request(self, request):
                     request.body = body.replace(old, new)
 
         return request
+
+
+def retry(func, timeout, interval=10):
+    """
+    Retry func until success.
+    :param timeout: in seconds
+    :param func: function
+    :param interval: polling interval
+    :return:
+    """
+    for i in range(0, timeout, interval):
+        try:
+            return func()
+        except Exception:  # pylint: disable=broad-except
+            time.sleep(interval)
```

Comment:
This method serves exactly the same functionality as https://github.com/Azure/azure-cli/blob/11347b73999dc9b9d1c6f924fb42e3d20b34a767/src/azure-cli/azure/cli/command_modules/role/tests/util.py#L12-L26